src/integrations/kotlin/nomic/data/repositories/games/GameRepositoryTest.kt

```diff
@@ -0,0 +1,107 @@
+package nomic.data.repositories.games
+
+import nomic.data.dtos.GameDTO
+import nomic.data.dtos.Games
+import nomic.data.dtos.games
+import nomic.domain.entities.EndUser
+import nomic.domain.entities.GameModel
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.MethodOrderer
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestMethodOrder
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import org.ktorm.database.Database
+import org.ktorm.dsl.eq
+import org.ktorm.entity.EntitySequence
+import org.ktorm.entity.filter
+import org.ktorm.entity.map
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.mock
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.MOCK
+)
+class GameRepositoryTest(@Autowired private val db: Database) {
+
+    private companion object {
+        private val user1: EndUser = EndUser(50, "ListGamesUser1")
+        private val user2: EndUser = EndUser(55, "ListGamesUser2")
+    }
+
+    @Test
+    fun test_listGames_dataFromDatabase() {
+        val repo = GameRepository(db)
+        val expectedGames = db.games.map { GameModel(it.gameId, it.title, it.createDate, it.currentPlayer, it.user.id) }
+            .associateBy { it.gameId }.toMutableMap()
+
+        val games = repo.listGames()
+
+        for (game in games) {
+            Assertions.assertThat(expectedGames).containsKey(game.gameId)
+            val expectedGame = expectedGames[game.gameId]!!
+
+            Assertions.assertThat(game.createDate).isEqualTo(expectedGame.createDate)
+            Assertions.assertThat(game.userId).isEqualTo(expectedGame.userId)
+            Assertions.assertThat(game.title).isEqualTo(expectedGame.title)
+            Assertions.assertThat(game.currentPlayer).isEqualTo(expectedGame.currentPlayer)
+
+            expectedGames.remove(game.gameId)
+        }
+
+        Assertions.assertThat(expectedGames.size).isZero()
+    }
+
+    @Test
+    fun test_listGames_callsSpecificationsInOrder() {
+        val repo = GameRepository(db)
+        val specUser = IGameSpecification { games -> games.filter { it.userId eq 1601 } }
+
+        var hasFirstSpecRun = false
+
+        val firstSpec = IGameSpecification {
+                games ->
+            hasFirstSpecRun = true
+            games
+        }
+
+        val secondSpec = IGameSpecification {
+                games ->
+            Assertions.assertThat(hasFirstSpecRun).isTrue
+            games
+        }
+
+        val games = repo.listGames(specUser, firstSpec, secondSpec)
+    }
```

Comment:
Gonna need an explanation on this test.
What is is doing? 
What is the thing you are testing for?